### PR TITLE
Forbid grant root user

### LINKED ISSUE #3235: Root user should not be reassigned other roles in space
**Describe the bug (__must be provided__)**
At now root user can be assigned to other roles in a space, it's confused for user which the real access he owns in the space? 

Accoding to my test, if I'm the root user of one nebula db which includes a space `test`
```
$ GRANT ROLE User ON test TO root
```
SHOW ROLES IN test; It's seems that I'm the `user` role of the space `test`, but I can modified the schema becaused of the db root role.

 User Role in space
image

`Create TAG` is allowed
image

**Additional context**
According to the confusing behavior of the role managing, it may cause some problems in applications about nebula.🤝

### PR DESCRIPTION
**Pull request by jackwener**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [x] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/graph/validator/test/ACLValidatorTest.cpp [hunk 1]
```diff
@@ -142,24 +142,6 @@ TEST_F(ACLValidatorTest, Simple) {
     ASSERT_EQ(*changePassword->newPassword(), newPassword);
   }
 
-  // grant role
```
Comment (Shylock-Hg) [inline src/graph/validator/test/ACLValidatorTest.cpp:145]: Why delete this?

File: src/graph/validator/ACLValidator.cpp [hunk 2]
```diff
@@ -104,10 +104,18 @@ Status ChangePasswordValidator::toPlan() {
 // grant role
 Status GrantRoleValidator::validateImpl() {
   const auto *sentence = static_cast<const GrantSentence *>(sentence_);
-  if (sentence->getAccount()->size() > kUsernameMaxLength) {
+  const std::string *account = sentence->getAccount();
+  if (account->size() > kUsernameMaxLength) {
     return Status::SemanticError("Username exceed maximum length %ld characters.",
                                  kUsernameMaxLength);
   }
+  auto metaClient = qctx_->getMetaClient();
```
Comment (Shylock-Hg) [inline src/graph/validator/ACLValidator.cpp:112]: You could get roles from session directly.

**Comment by Shylock-Hg** [comment]
Need add test case in tck.

File: src/graph/validator/ACLValidator.cpp (hunk 2 above)
Comment (Shylock-Hg) [reply src/graph/validator/ACLValidator.cpp]: Sorry, it's my mistake, the granted user is `Account` from sentence and is different from current user of session.

File: src/graph/validator/test/ACLValidatorTest.cpp (hunk 1 above)
Comment (jackwener) [reply src/graph/validator/test/ACLValidatorTest.cpp:145]: Because there is metaclient in this function.

Comment (Shylock-Hg) [reply src/graph/validator/test/ACLValidatorTest.cpp:145]: This case should keep.

File: src/graph/validator/test/ACLValidatorTest.cpp [hunk 3]
```diff
@@ -142,24 +142,6 @@ TEST_F(ACLValidatorTest, Simple) {
...
-  {
```
Comment (Shylock-Hg) [inline src/graph/validator/test/ACLValidatorTest.cpp:146]: This case should keep.

**Review by Shylock-Hg — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Review by Shylock-Hg — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**